Add Userbase guide (#1775)
vercel/docs: pages/guides/deploying-userbase-with-vercel.mdx

import Guide from '~/components/layout/guide'
import { Code } from '~/components/code'
import Snippet from '~/components/snippet'
import Caption from '~/components/text/caption'
import Link from '~/components/text/link'
import { Image } from '~/components/media'

> import DeploySection from '~/components/guides/deploy-section'

Reviewer: ```suggestion
import DeploySection from '~/components/guides/deploy-section'
import { InlineCode } from '~/components/text/code'
```

import NameWrapper from '~/components/name/name-wrapper'
import { PRODUCT_NAME } from '~/lib/constants'
import {
  AppJs,
  Layout,
  Nav,
  Modal,
  TodoForm,
  HomePage,
  ButtonStyles,
  IndexCSS

> } from '~/lib/deploy-userbase-guide-code-fragments'

Reviewer: Can these be removed now?

export const meta = {

>   title: `Deploy Userbase with ${PRODUCT_NAME}`,

Reviewer: ```suggestion
  title: `Deploy a Userbase Secured Application with ${PRODUCT_NAME}`,
```

  description: `Create a todo app with Userbase and Next.js and deploy it live with ${PRODUCT_NAME}.`,
  published: '2020-04-22T12:00:00.860Z',
  authors: ['coetry'],

>   url: '/guides/deploying-userbase-with-vercel',

Reviewer: Has this been added to Checkly?

  demo: 'https://next-userbase.now.sh',
  image: '',
  editUrl: 'pages/guides/deploying-userbase-with-vercel.mdx',
  lastEdited: '2020-04-23T21:21:57.000Z'
}

[Userbase](https://userbase.com/) is the easiest way to add user accounts and data persistence to your static site.

> All Userbase features are accessible through a very simple [JavaScript SDK](https://userbase.com/docs/sdk/), directly from the browser without a backend.

Reviewer: ```suggestion
All Userbase features are accessible through a [JavaScript SDK](https://userbase.com/docs/sdk/), directly from the browser, without requiring a backend.
```

> This guide walks through building a Todo app with Userbase and Next.js.

Reviewer: ```suggestion
This guide walks you through building a todo app with Userbase and Next.js, and how to deploy it with Vercel.
```

## Step 1: Create A Userbase Account

> First you'll need to create a Userbase account if you don't already have one. You can create one in _less than 10 seconds_ [here](https://v1.userbase.com/#create-admin).

Reviewer: ```suggestion
First, you'll need to create a Userbase account if you don't already have one. You can create one in _less than 10 seconds_ [here](https://v1.userbase.com/#create-admin).
```

> After creating your account, make note of your _App ID_. We'll store it later as an environment variable for our project.

Reviewer: ```suggestion
After creating your account, make a note of your _App ID_. You'll store it later as an environment variable for your project.
```

> ## Step 2: Set Up A New Next.js Project

Reviewer: ```suggestion
## Step 2: Set up a New Next.js Project
```

> Let's start off by scaffolding a Next.js project with [Tailwind](https://tailwindcss.com/) for quick styling.

Reviewer: ```suggestion
Set up a Next.js and Userbase project by initializing the [project template](/path-to-example) and moving into the project directory:
```

<Snippet dark text="npm init next-app --example with-userbase userbase-todo-app && cd userbase-todo-app" />

> <Caption>Initializing a Next.js + Userbase app and entering the new project directory</Caption>

Reviewer: ```suggestion
<Caption>Initializing a Next.js + Userbase app and entering into the project directory.</Caption>
```

## Step 3: Add Environment Variables

> The scaffolded template expects us to provide a `USERBASE_APP_ID` variable as shown in `next.config.js`.

Reviewer: ```suggestion
The scaffolded template expects you to provide a `USERBASE_APP_ID` variable, as shown in the `next.config.js` file.
```

<Code lang="js">
  {`
    module.exports = {
      env: {
        USERBASE_APP_ID: process.env.USERBASE_APP_ID,
      },
    };
`}
</Code>

> <Caption>`USERBASE_APP_ID` environment variable in next.config.js</Caption>

Reviewer: ```suggestion
<Caption>The <InlineCode>next.config.js</InlineCode> file that makes the <InlineCode>USERBASE_APP_ID</InlineCode> environment variable available to the application.</Caption>
```

> For development, you can simply add your Userbase _App ID_ from _Step 1_ as an environment variable in a `.env` file in your project root and run `next dev` to start a development server.

Reviewer: ```suggestion
For development, you can add your Userbase _App ID_ from _Step 1_ as an environment variable in a `.env` file at the project root and run `next dev` to start a development server.
```

<Code lang="bash">
  {`USERBASE_APP_ID=<paste-your-id-here>`}
</Code>

> <Caption>`USERBASE_APP_ID` environment variable defined in .env</Caption>

Reviewer: ```suggestion
<Caption>Definining the <InlineCode>USERBASE_APP_ID</InlineCode> environment variable in a <InlineCode>.env</InlineCode> file.</Caption>
```

In order to take your app to production, you can add the _App ID_ as a secret by running the following command:

<Snippet dark text="now secrets add userbase-app-id [paste-your-id-here]" />

> <Caption>Adding a secret to your Vercel account. Make sure to remove the brackets.</Caption>

Reviewer: ```suggestion
<Caption>Adding a secret to your Vercel account.</Caption>
```

> Our template already provides us with a `now.json` that makes use of the secret we added above.

Reviewer: ```suggestion
The project template already provides you with a `now.json` file that makes use of the secret added above.
```

<Code lang="json">
  {`
    {
      "build": {
        "env": {
          USERBASE_APP_ID: "@userbase-app-id"
        }
      }
    }
  `}
</Code>

> <Caption>`now.json` reading our secret as a build environment variable</Caption>

Reviewer: ```suggestion
<Caption>The <InlineCode>now.json</InlineCode> file that provides the defined secret to the application.</Caption>
```
